cluster-autoscaler/clusterstate/clusterstate.go

```diff
@@ -271,6 +279,13 @@ func (csr *ClusterStateRegistry) updateScaleRequests(currentTime time.Time) {
 				nodeGroupName, currentTime.Sub(scaleUpRequest.Time))
 			continue
 		}
+		_, registrationTime, err := csr.GetProvisioningTimeouts(scaleUpRequest.NodeGroup)
+		if err != nil {
+			klog.Warningf("failed to retrieve max node registering time for nodeGroup %s", nodeGroupName)
+			continue
+		} else if csr.allInstancesAreCreated(nodeGroupName) && currentTime.Add(registrationTime).Before(scaleUpRequest.ExpectedAddTime) {
+			scaleUpRequest.ExpectedAddTime = currentTime.Add(registrationTime)
+		}
```

Comment:
This feels extremely hacky. IIUC we want to update the `ExpectedAddTime` once all the nodes are created to handle the separate registration time limit.

Can we just add a different variable to `scaleUpRequest` that would be initially 0 and then set once the nodes are created? IMO we should rename the `ExpectedAddTime` to `ExpectedProvisionTime` and add a new `ExpectedRegistrationTime` variable that would be dedicated to registration limits. Otherwise this becomes really unreadable.